cedar-policy-core/src/ast/request.rs

```diff
@@ -49,6 +49,17 @@ pub struct Request {
     pub(crate) context: Option<Context>,
 }
 
+/// Represents the principal type, resource type, and action UID.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize, Serialize)]
+pub struct RequestType {
```

Comment:
Why do you need to define this type here if we already have `RequestEnv` in the validator code?

Reply:
As far as I can tell, entity manifests need less information than `RequestEnv` because we validate the policies in strict mode. This helps prevent some runtime errors (hopefully).